demos/link-prediction/random-walks/ctdne-link-prediction.ipynb

```diff
@@ -0,0 +1,680 @@
+    "        )\n",
+    "    )\n",
+    "    return random.choices(possible_edges_neg, k=len(positive_examples))\n",
+    "\n",
+    "\n",
+    "pos, neg = create_link_examples(graph, edges_train)\n",
+    "pos_test, neg_test = create_link_examples(graph, edges_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It's good to verify that the data structures we've created so far from the raw data look reasonable.\n",
+    "\n",
+    "`StellarGraph.info` is a useful method for inspecting the graph we've created, and we can also check that the number of link examples correspond to the train/test subset values we defined earlier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "StellarGraph: Undirected multigraph\n",
+      " Nodes: 151, Edges: 37929\n",
+      "\n",
+      " Node types:\n",
+      "  default: [151]\n",
+      "    Features: none\n",
+      "    Edge types: default-default->default\n",
+      "\n",
+      " Edge types:\n",
+      "    default-default->default: [37929]\n",
+      "\n",
+      "Training examples: 9524 positive links, 9524 negative links\n",
+      "Validation examples: 3119 positive links, 3119 negative links\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(graph.info())\n",
+    "print()\n",
+    "print(f\"Training examples: {len(pos)} positive links, {len(neg)} negative links\")\n",
+    "print(\n",
+    "    f\"Validation examples: {len(pos_test)} positive links, {len(neg_test)} negative links\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running random walks\n",
+    "\n",
+    "We try to keep the setup comparable between the use of temporal and biased (static) random walks. For temporal walks, the input parameter is defined in terms of the total number of context windows you are interested in obtaining, which differs from the traditional approach of specifying the number of walks to run per node in the graph. We calculate the number of context windows we need in terms of the traditional parameters as:\n",
+    "```\n",
+    "num_cw = num_nodes * num_walks_per_node * (walk_length - cw_size)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of temporal random walks: 1763\n",
+      "Number of static random walks: 1510\n"
+     ]
+    }
+   ],
+   "source": [
+    "from stellargraph.data import TemporalRandomWalk, BiasedRandomWalk\n",
+    "\n",
+    "num_walks_per_node = 10\n",
+    "walk_length = 80\n",
+    "cw_size = 10\n",
+    "num_cw = (\n",
+    "    len(graph.nodes()) * num_walks_per_node * (walk_length - cw_size)\n",
+    ")  # as recommended in reference paper\n",
```

Comment:
This line has a lot of moving parts, so what specifically was recommended in the reference paper? Maybe this could refer to a specific section?